docs/GettingStartedDocs/OP-TEE/Debugging/QEMU.md

```diff
@@ -0,0 +1,412 @@
+boot must be generated. This section shows you how to retrieve all the software
+components required, how to build them, and how to run them in QEMU.
+
+## Sources
+
+The `repo` utility can manipulate multiple Git repositories as though they were
+one. The following commands instruct `repo` to clone all the repositories that
+are required to build a QEMU-based debugging environment:
+
+```
+[ TERM 1 ]
+
+mkdir emulation
+cd emulation
+
+repo init -u https://github.com/ms-iot/optee_manifest -m oe_qemu_v8.xml -b oe-3.6.0
+repo sync
+```
+
+`repo sync` takes some time, seeing as it clones the Linux kernel, among other
+things. To clone multiple repositories in parallel, add the `-j` switch to the
+`repo sync` command in the same way you would an invocation of `make`.
+
+## Building
+
+To create and launch a debugging environment all you need is `make`. Depending
+on your machine, this may take upward of an hour the first time; subsequent runs
+only take a few seconds, plus compiling anything that you may have changed:
+
+```
+[ TERM 1 ]
+
+cd build
+
+make toolchains -j2
+make run -j$(nproc)
+```
+
+**Note**: `make toolchains` need only be called once, and since it downloads two
+files, the `-j2` is already included.
+
+Once this command is complete, it launches QEMU inside the same terminal as
+where you executed `make run` along with two XTerm windows. In the terminal, you
+can control QEMU via its monitor interface. For example, typing `c <Enter>`
+resumes execution and `q <Enter>` quits QEMU.
+
+The two XTerm windows are connected to one emulated serial port each: one shows
+output from the Normal World (the REE, or untrusted side) and the other shows
+output from the Secure World (the TEE, or trusted side). The Normal World XTerm
+window allows you to interact with the emulated Linux environment through a
+BusyBox shell.
+
+By default, the emulated processor is halted when QEMU starts. To ensure that
+the build is sane, resume execution by issuing the `c <Enter>` command in the
+QEMU monitor (`c` is short for "continue").
+
+In the Secure World XTerm window you should see OP-TEE's initialization output.
+In the Normal World XTerm window you should see Linux boot. Try logging into
+Linux in the Normal World XTerm window once it is done booting (see the output
+in XTerm on how to log in).
+
+## Emulator Setup
+
+To debug enclaves on ARM TrustZone, you must build host applications and
+enclaves. Then, you must copy your enclaves into it.
+
+### Building Hosts & Enclaves
+
+```
+[ TERM 2 ]
+
+git clone --recursive https://github.com/openenclave/openenclave.git sdk
+
+cd sdk
+
+sudo scripts/ansible/install-ansible.sh
+sudo ansible-playbook scripts/ansible/oe-contributors-setup-cross-arm.yml
+
+cd ..
+
+mkdir build
+cd build
+
+cmake ../sdk \
+    -G Ninja \
+	-DHAS_QUOTE_PROVIDER=OFF \
+	-DCMAKE_TOOLCHAIN_FILE=../sdk/cmake/arm-cross.cmake \
+	-DOE_TA_DEV_KIT_DIR=$PWD/../emulation/optee_os/out/arm/export-ta_arm64 \
```

Comment:
This path is different than the one I left a comment about above, and looks more like a real path that should exist! Is that okay? :)

Reply:
This one is good. The path in this command makes use of the build output that was generated in previous steps.